### PR TITLE
New bot, aka. Robocoup Mk. II. Almost done.

### PR DESCRIPTION
**Pull request by cowboy**
* Includes everything but expertise update dialog support.
* Uses `TOKEN_NEWBOT` for now.

### CODE REVIEW

**Comment by tkellen** [comment]
This implementation looks very clean, but I think the low level abstractions you've made might do with a bit of sugar. I wonder if we couldn't present a slightly more high level API via the slackbot adapter we talked about yesterday? Let's chat about it during our one-on-one to see if that makes any sense.

**Comment by cowboy** [comment]
That's what I was planning on doing. Note the WIP!

**Comment by cowboy** [comment]
@jennschiffer @tkellen any questions before I merge?

File: src/new-bot/commands/out.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/*
+ * A command to show who is out today.
+ */
+import moment from 'moment';
+import {createCommand} from 'chatter';
+import {query} from '../../lib/db';
+
+export const usage = 'Show who is out today:\n`Usage: out`';
```
Comment (cowboy) [inline src/new-bot/commands/out.js]: TODO: remove

File: src/new-bot/index.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      webClient: new WebClient(config.tokens.newbot),
+    };
+  },
+  createMessageHandler(id, {channel}) {
+    // Direct message
+    if (channel.is_im) {
+      return createConversation([
+        // Nameless command that encapsulates sub-commands and adds a "help"
+        // command and a fallback message handler.
+        lolHandler,
```
Comment (tkellen) [inline src/new-bot/index.js]: do we need lolHandler still?

File: src/new-bot/commands/status.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  usage: '[me | @bocouper]',
+}, createParser(({remain}, {user}) => {
+  let [slackname] = remain;
+  if (!slackname) {
+    return false;
+  }
+  if (slackname === 'me') {
+    slackname = user.name;
+  }
+  return query('status', slackname).get(0).then(status => {
```
Comment (tkellen) [inline src/new-bot/commands/status.js]: status me doesn't work, the username isn't being passed correctly.

**Comment by tkellen** [comment]
:+1: to merge with minor fixes

**Pull request merged**